crates/rolldown_plugin_isolated_declaration/src/lib.rs

```diff
@@ -58,7 +56,7 @@ impl Plugin for IsolatedDeclarationPlugin {
       let codegen_ret =
         CodeGenerator::new().with_options(CodegenOptions::default()).build(&ret.program);
 
-      let mut emit_dts_path = Path::new(args.id).to_path_buf();
+      let mut emit_dts_path = Path::new(args.stable_id).to_path_buf();
```

Comment:
Would it be better to calculate itself?

Stable id might be not valid path. It may adds more non-path content to keep uniqueness.